shared/src/commonMain/kotlin/co/touchlab/kampkit/viewmodel/BreedViewModelInterface.kt

```diff
@@ -1,51 +1,48 @@
-package co.touchlab.kampkit.android
+package co.touchlab.kampkit.viewmodel
 
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
 import co.touchlab.kampkit.db.Breed
 import co.touchlab.kampkit.models.BreedModel
 import co.touchlab.kampkit.models.DataState
 import co.touchlab.kampkit.models.ItemDataSummary
 import co.touchlab.kermit.Kermit
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.flattenMerge
 import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
-import org.koin.core.component.inject
-import org.koin.core.parameter.parametersOf
 
-class BreedViewModel : ViewModel(), KoinComponent {
+interface BreedViewModelInterface : KoinComponent {
 
-    private val log: Kermit by inject { parametersOf("BreedViewModel") }
-    private val scope = viewModelScope
-    private val breedModel: BreedModel = BreedModel()
-    private val _breedStateFlow: MutableStateFlow<DataState<ItemDataSummary>> = MutableStateFlow(
-        DataState(loading = true)
-    )
+    var scope: CoroutineScope
+    val log: Kermit
+    val breedModel: BreedModel
 
-    val breedStateFlow: StateFlow<DataState<ItemDataSummary>> = _breedStateFlow
-
-    init {
+    fun initWithScope(newScope: CoroutineScope) {
+        this.scope = newScope
         observeBreeds()
     }
 
+    fun getFlowValue(): DataState<ItemDataSummary>
+    fun setFlowValue(value: DataState<ItemDataSummary>)
+
+    val breedStateFlow: StateFlow<DataState<ItemDataSummary>>
+
     @OptIn(FlowPreview::class)
-    private fun observeBreeds() {
+    fun observeBreeds() {
```

Comment:
is it possible to pull the implementations into the `SharedBreedViewModel` class? Using default interface implementations feels weird for logic

Reply:
Yep, that would definitely work.